### DISCUSSION TITLE
Remeshing during simulation

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#21631] [Q&A Meshing] [answered]
Sometime in the future, I want to implement a remesher for when the mesh gets deformed. For a first go around, I will use the built-in libMesh::LaplaceMeshSmoother. If someone could give an initial list of things I would need to think about that would be helpful. For example interpolate current material properties onto the new mesh, etc. Let me know if there are any examples in moose that would be helpful also. I will be diving deeper in this later in the future, but I wanted to get some feedback and hints in the meantime! Thanks for any help.

**Comment by hugary1995** [comment, marked as answer]
The list of things to consider:
1. projecting variables
2. projecting material properties
3. projecting stateful material properties
4. remeshing part of the domain, e.g. subdomain-restricted remeshing
5. iterative remeshing, e.g. your remeshing criteria may change after you remesh, so you may want to recompute the remeshing criteria which is possibly followed by another remeshing...
6. preserving locations of nodesets, sidesets, interfaces within a subdomain
7. optionally re-equilibriating the system after remeshing, which, again, may be followed by another remeshing in an iterative fashion.

For a prototype, I think 1 and 2 with a simple remeshing criteria is sufficient.

I wonder if the MOOSE team and/or Coreform already have projects along this line. You may want to check with them first.

**Comment by GiudGiud** [comment]
Hello

So with the current capability there are essentially two ways I would do this:

- write down all the mesh changing capability needed in a user object and execute this user object only when needed. You may use the Controls system to do that, to execute it exactly when you want to. This would be done in a single calculation/single input file approach.

OR

- write all the necessary data (all field variables etc) to a restart file, either exodus or checkpoint. Then you will NOT be able to restart directly, but you may restart this data in a MultiApp executed on INITIAL with the original mesh then move the data over using transfers (which will take care of projections). This is a little obnoxious but our only work around for the element renumbering that would occur during remeshing

Note: If you are ONLY doing smoothing and the elements are the same, you may consider restarting directly

In the future (eg some time next FY) we will be able run mesh generators during a calculation.

Guillaume
